types/web3-provider-engine.d.ts

```diff
@@ -1 +1,37 @@
-declare module 'web3-provider-engine';
+declare module 'web3-provider-engine' {
+  import type { EventEmitter } from 'events';
+
+  type RpcPayload<P> = {
+    id: unknown;
```

Comment:
These types aren't comprehensive, otherwise this would be more exact. My goal was to look at the JavaScript code and inferred the minimum set of types based on what I saw.